test/framework/toolchain.py

```diff
@@ -948,6 +948,32 @@ def test_nosuchtoolchain(self):
         tc = self.get_toolchain('intel', version='1970.01')
         self.assertErrorRegex(EasyBuildError, "No module found for toolchain", tc.prepare)
 
+    def test_mpi_exec_nranks(self):
+        """Test mpi_exec_nranks function."""
+        self.modtool.prepend_module_path(self.test_prefix)
+
+        tc = self.get_toolchain('gompi', version='2018a')
+        tc.prepare()
+        self.assertEqual(tc.mpi_exec_nranks(nr_ranks=2), "mpirun -n 2")
+        self.modtool.purge()
+
+        self.setup_sandbox_for_intel_fftw(self.test_prefix)
+        tc = self.get_toolchain('intel', version='2018a')
+        tc.prepare()
+        self.assertEqual(tc.mpi_exec_nranks(nr_ranks=2), "mpirun -n 2")
+        self.modtool.purge()
+
+        self.setup_sandbox_for_intel_fftw(self.test_prefix, imklver='10.2.6.038')
+        tc = self.get_toolchain('intel', version='2012a')
+        tc.prepare()
+
+        mpi_exec_nranks_re = re.compile("^mpirun --file=.*/mpdboot -machinefile .*/nodes -np 4")
+        self.assertTrue(mpi_exec_nranks_re.match(tc.mpi_exec_nranks(nr_ranks=4)))
+
+        # test specifying custom template for MPI commands
+        init_config(build_options={'mpi_exec_template': "mpiexec -np %(nr_ranks)s --", 'silent': True})
+        self.assertEqual(tc.mpi_exec_nranks(nr_ranks="7"), "mpiexec -np 7 --")
```

Comment:
Also check default `tc.mpi_exec_nranks()`

Passing a string argument '7' here only works because your custom template uses `%(nr_ranks)s` rather than `%(nr_ranks)d`.

It probably makes sense to just also use `%(nr_ranks)s` in the default template, so both `7` and `'7'` can be passed as an argument to `mpi_exec_nranks`...